Assets/SEE/UI/Window/PropertyWindow/PropertyWindow.cs

```diff
@@ -0,0 +1,845 @@
+                        }
+                        key = InsertDotInFirstPascalCase(pair.Key);
+                        /// To remove duplicates it is needed to remove the old one. <see cref="GraphWriter"/> L.313.
+                        if (attributes.ContainsKey(key) && key.Contains("Source.Range"))
+                        {
+                            attributes.Remove(key);
+                        }
+                        attributes.Add(key, pair.Value);
+                    }
+                }
+                if (GraphElement.FloatAttributes.Count > 0 & contextMenu.Filter.IncludeFloatAttributes)
+                {
+                    foreach (KeyValuePair<string, float> pair in GraphElement.FloatAttributes)
+                    {
+                        attributes.Add(InsertDotInFirstPascalCase(pair.Key), pair.Value);
+                    }
+                }
+                SplitInAttributeGroup(attributes);
+
+                return;
+                string InsertDotInFirstPascalCase(string input)
+                {
+                    /// Regular Expression Pattern
+                    string pattern = @"^([A-Z][a-z]+)([A-Z][a-z]+)(_.*)$";
+
+                    Regex regex = new(pattern);
+                    Match match = regex.Match(input);
+
+                    if (match.Success)
+                    {
+                        /// Build the new string by inserting a period between the matched groups
+                        return $"{match.Groups[1].Value}.{match.Groups[2].Value}{match.Groups[3].Value}";
+                    }
+                    else
+                    {
+                        /// Return the original input if it doesn't match the pattern
+                        return input;
+                    }
+                }
+            }
+        }
+
+        #region Grouping Name Type
+        /// <summary>
+        /// Divides the attributes into groups and subgroups.
+        /// </summary>
+        /// <param name="attributes">The attributes to divide.</param>
+        private void SplitInAttributeGroup(Dictionary<string, object> attributes)
+        {
+            Dictionary<string, object> nestedDict = new();
+            foreach (KeyValuePair<string, object> pair in attributes)
+            {
+                AddNestedAttribute(nestedDict, pair.Key.Split('.'), pair.Value);
+            }
+            CreateNestedGroups(nestedDict);
+        }
+
+        /// <summary>
+        /// Inserts an attribute to the nested dictionary.
+        /// </summary>
+        /// <param name="dict">The nested dictionary.</param>
+        /// <param name="keys">The keys to add.</param>
+        /// <param name="value">The value of the attribute.</param>
+        private void AddNestedAttribute(Dictionary<string, object> dict, string[] keys, object value)
+        {
+            Dictionary<string, object> currentDict = dict;
+
+            for (int i = 0; i < keys.Length - 1; i++)
```

Comment:
This could iterate over the keys using `foreach`, right?

Reply:
No, it is not a `foreach`. The last element of the keys array is the attribute name to which the value should be assigned. This key is not a group.